modules/apps/transfer/types/transfer_authz.go

```diff
@@ -0,0 +1,116 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+
+	"golang.org/x/exp/slices"
+)
+
+const gasCostPerIteration = uint64(10)
+
+var (
+	_ authz.Authorization = &TransferAuthorization{}
+)
+
+// NewTransferAuthorization creates a new TransferAuthorization object.
+func NewTransferAuthorization(sourcePorts, sourceChannels []string, spendLimits []sdk.Coins, allowedAddrs [][]string) *TransferAuthorization {
+	allocations := []PortChannelAmount{}
+	for index := range sourcePorts {
+		allocations = append(allocations, PortChannelAmount{
+			SourcePort:       sourcePorts[index],
+			SourceChannel:    sourceChannels[index],
+			SpendLimit:       spendLimits[index],
+			AllowedAddresses: allowedAddrs[index],
+		})
+	}
+	return &TransferAuthorization{
+		Allocations: allocations,
+	}
+}
+
+// MsgTypeURL implements Authorization.MsgTypeURL.
+func (a TransferAuthorization) MsgTypeURL() string {
+	return sdk.MsgTypeURL(&MsgTransfer{})
+}
+
+func IsAllowedAddress(ctx sdk.Context, receiver string, allowedAddrs []string) bool {
```

Comment:
We could make this function unexported/private.

Should we consider supporting allow all receivers functionality by using a sentinel const value similarly to how we support [`AllowAllHostMsgs`](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/host/types/keys.go#L15) in interchain accounts, or by leaving this list empty?

cc. @womensrights @crodriguezvega